### ISSUE TITLE
Add ability to switch among different groups of tabs in the tab bar

### ISSUE THREAD

**Issue by rdong8** [brave/brave-browser#40517]
### Platforms

all

### Description

There is a Firefox addon called [Simple Tab Groups](https://addons.mozilla.org/en-CA/firefox/addon/simple-tab-groups/) that is the only thing keeping me and [many other](https://www.reddit.com/r/chrome/comments/9ucckv/i_use_firefox_just_because_1_addon_does_chrome/) Firefox users from switching to Brave. All it does is provide tab groups (basically a saved window accessible through a switcher panel):

![image](https://github.com/user-attachments/assets/36a86ab9-0f1c-4e44-9dfa-415298a32334)

![image](https://github.com/user-attachments/assets/c756779f-8af3-445c-a307-b94c07c0b5a9)

There are no Chrome extensions that provide the equivalent behaviour - the most they do is display Brave's native tab groups in a vertical panel, but no actual separation is provided between different groups as is present in the Simple Tab Groups extension. For instance, CTRL + Tab will traverse between tabs of different groups with the other extensions, which is not the case with Simple Tab Groups.

**Comment by rebron** [comment]
@rdong8 We have tab groups in vertical tabs. See the attached screenshot. Just right click on a tab to create a group.

The way that Simple Tab Groups add-ons works is more like a "Spaces" concept. It looks like when choosing a group say "Work" it would replace whatever is active, like "Finance" right?

<img width="270" alt="Screenshot 2024-08-18 at 9 21 38 PM" src="https://github.com/user-attachments/assets/f489c355-48ac-4909-99ec-ee78ae7728be">

**Comment by rdong8** [comment]
Yes I meant something more similar to spaces, I just want more isolation between the different groups. Even with vertical tabs, the different tab groups still show up in the tab switcher.

**Comment by rebron** [comment]
<img width="427" height="188" alt="Image" src="https://github.com/user-attachments/assets/6e478a5e-34a3-4261-b21f-e28e9ce64ad8" />

<img width="346" height="265" alt="Image" src="https://github.com/user-attachments/assets/81f7d982-aca8-470f-b966-9cdc88a3ca38" />

@rdong8 Closing this as wfm. This is the Saved Tab Groups feature for the bookmarks bar and gives you the ability to view tab groups and open in a new window. We can reopen if this doesn't work for you.

**Comment by rdong8** [comment]
The issue should be reopened, this is not the same feature. I've attached a screen recording to make it more clear. In the Simple Tab Groups extension, the tab groups do not open in separate windows - you can switch between them within the same window instantaneously by clicking on the switcher.

https://github.com/user-attachments/assets/5d6c3745-d595-4dff-9cb1-edda4086e42d

**Comment by rebron** [comment]
@rdong8 Can you confirm, when you select and open up a tab group, it add them to the right in the same window? If that's the case, that what happens now when you select a tab group. The tab group will open in the same window. Is it an add or a replace? If replace then you are talking about more of a spaces feature.

<img width="423" height="198" alt="Image" src="https://github.com/user-attachments/assets/036b35e3-24bc-40ff-b167-24bd8f2614f8" />

**Comment by rdong8** [comment]
I think what Firefox itself and Brave refer to as tab groups are just the literal groupings of tabs in the tab bar. The Simple Tab Groups extension allows those native browser tabs to be used in conjunction with the spaces it provides and refers to as tab groups. I find the latter to be far more useful since the addition of more groups does not shrink the space in the tab bar, and they're isolated from each other.
